dll/win32/browseui/shellbars/CBandSiteMenu.cpp

```diff
@@ -48,19 +48,6 @@ CBandSiteMenu::~CBandSiteMenu()
 
 HRESULT WINAPI CBandSiteMenu::FinalConstruct()
 {
-    HRESULT hr = SHGetFolderLocation(0, CSIDL_DESKTOP, NULL, 0, &m_DesktopPidl);
-    if (FAILED_UNEXPECTEDLY(hr))
-        return hr;
-
-    WCHAR buffer[MAX_PATH];
-    hr = SHGetFolderPathAndSubDirW(0, CSIDL_APPDATA | CSIDL_FLAG_CREATE, NULL, 0, L"Microsoft\\Internet Explorer\\Quick Launch", buffer);
-    if (FAILED_UNEXPECTEDLY(hr))
-        return hr;
-
-    m_QLaunchPidl.Attach(ILCreateFromPathW(buffer));
-    if (m_QLaunchPidl == NULL)
-        return E_FAIL;
-
```

Comment:
Please not that _GetMenuIdFromISFBand uses m_QLaunchPidl as well so it won't have the correct behaviour. Also, after it is initialized once, there is no need to free it again.

Reply:
> Please not that _GetMenuIdFromISFBand uses m_QLaunchPidl as well so it won't have the correct behaviour.

Indeed, thanks, additional check added :)

> Also, after it is initialized once, there is no need to free it again.

When E_FAIL returned at `FinalConstruct`, "Toolbars" menu wouldn't be created at all. It can fail only when path doesn't exist, and no `CSIDL_FLAG_CREATE` flag specified.

Windows doesn't create Quick Launch directory:
- when it doesn't exist, "Toolbars" menu is still accessible
- but once Quick Launch toolbar gets called, it should throw error "Cannot create toolbar."

If I initialize it once, it wouldn't throw failure at `_CreateBuiltInISFBand` when I call the toolbar next time (of course, if "Quick Launch" directory doesn't exist anymore).